src/Compiler/Microsoft.AspNetCore.Razor.Language/test/TestFiles/IntegrationTests/ComponentRuntimeCodeGenerationTest/AsyncEventHandler_OnElement_ActionEventArgs_Lambda/TestComponent.mappings.txt

```diff
@@ -0,0 +1,17 @@
+﻿Source Location: (1:0,1 [30] x:\dir\subdir\Test\TestComponent.cshtml)
+|using System.Threading.Tasks
+|
+Generated Location: (317:11,0 [30] )
+|using System.Threading.Tasks;
|

```

Comment:
The source location includes a newline not present in the generated location. Is that a problem?

I wonder if we should assert that source and generated location texts match so we don't have to check this manually.

Reply:
The generated location also includes a semi-colon. I think this would be a problem, yes.

> I wonder if we should assert that source and generated location texts match

I thought there was such a validation, maybe its just not called in all cases. Pretty sure I've hit it once or twice.

Reply:
Oh huh, good catch. I'll add the assertion if we don't have it, too.

Reply:
So, it turns out fixing the using discrepancy is significantly more involved, as it's actually a difference in parsing that's occurring.

I've backed out the `using` mappings from this PR, rather than try and fix a huge amount of stuff in one go.